src/instana/instrumentation/asyncio.py

```diff
@@ -2,46 +2,89 @@
+from opentelemetry.trace import use_span
+from opentelemetry.trace.status import StatusCode
 
-from ..configurator import config
-from ..log import logger
-from ..singletons import async_tracer
+from instana.configurator import config
+from instana.log import logger
+from instana.span.span import InstanaSpan
+from instana.util.traceutils import get_tracer_tuple, tracing_is_off
 
 try:
     import asyncio
 
     @wrapt.patch_function_wrapper("asyncio", "ensure_future")
-    def ensure_future_with_instana(wrapped, instance, argv, kwargs):
-        if config["asyncio_task_context_propagation"]["enabled"] is False:
-            with no_parent_scope():
-                return wrapped(*argv, **kwargs)
-
-        scope = async_tracer.scope_manager.active
-        task = wrapped(*argv, **kwargs)
-
-        if scope is not None:
-            setattr(task, ACTIVE_ATTR, scope)
+    def ensure_future_with_instana(
+        wrapped: Callable[..., asyncio.ensure_future],
+        instance: object,
+        argv: Tuple[object, Tuple[Any, ...]],
+        kwargs: Dict[str, Any],
+    ) -> object:
+        if (
+            config["asyncio_task_context_propagation"]["enabled"] is False
+            or tracing_is_off()
+        ):
+            return wrapped(*argv, **kwargs)
 
-        return task
+        with _start_as_current_async_span() as span:
+            try:
+                span.set_status(StatusCode.OK)
+                return wrapped(*argv, **kwargs)
+            except Exception as exc:
+                logger.debug(f"asyncio ensure_future_with_instana error: {exc}")
 
     if hasattr(asyncio, "create_task"):
 
         @wrapt.patch_function_wrapper("asyncio", "create_task")
-        def create_task_with_instana(wrapped, instance, argv, kwargs):
-            if config["asyncio_task_context_propagation"]["enabled"] is False:
-                with no_parent_scope():
+        def create_task_with_instana(
+            wrapped: Callable[..., asyncio.create_task],
+            instance: object,
+            argv: Tuple[object, Tuple[Any, ...]],
+            kwargs: Dict[str, Any],
+        ) -> object:
+            if (
+                config["asyncio_task_context_propagation"]["enabled"] is False
+                or tracing_is_off()
+            ):
+                return wrapped(*argv, **kwargs)
+
+            with _start_as_current_async_span() as span:
+                try:
+                    span.set_status(StatusCode.OK)
                     return wrapped(*argv, **kwargs)
+                except Exception as exc:
+                    logger.debug(f"asyncio create_task_with_instana error: {exc}")
 
-            scope = async_tracer.scope_manager.active
-            task = wrapped(*argv, **kwargs)
+    @contextmanager
+    def _start_as_current_async_span() -> Iterator[InstanaSpan]:
+        """
+        Creates and yield a special InstanaSpan to only propagate the Asyncio
+        context.
```

Comment:
Is there a specific reason why we don't use `NonRecordingSpan` provided by OTel?